tutorials/llm/llama-3/nemo2-sft-peft/nemo2-peft.ipynb

```diff
@@ -0,0 +1,540 @@
+    "    )\n",
+    "    return run.Config(\n",
+    "        nl.MegatronOptimizerModule,\n",
+    "        config=opt_cfg\n",
+    "    )\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### LoRA Adapter\n",
+    "We need to pass in LoRA adapter to our finetuning API to perform LoRA finetuning. We can configure adapter like the following. The target module we support includes: `linear_qkv`, `linear_proj`, `linear_fc1` and `linear_fc2`. In the final script we used default configurations for LoRA (`llm.peft.LoRA()`), which will use the full list with `dim=32`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def lora() -> run.Config[nl.pytorch.callbacks.PEFT]:\n",
+    "    return run.Config(LoRA)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Base Model\n",
+    "We will perform PEFT on top of Llama3-8b so we create a `LlamaModel` to pass to finetune API."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def llama3_8b() -> run.Config[pl.LightningModule]:\n",
+    "    return run.Config(llm.LlamaModel, config=run.Config(llm.Llama3Config8B))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### AutoResume\n",
+    "In NeMo 2.0 we can directly pass in Llama3-8b's Hugging Face ID to start PEFT without manually converting it into NeMo checkpoint format like in NeMo 1.0."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def resume() -> run.Config[nl.AutoResume]:\n",
+    "    return run.Config(\n",
+    "        nl.AutoResume,\n",
+    "        restore_config=run.Config(nl.RestoreConfig,\n",
+    "            path=\"nemo://meta-llama/Meta-Llama-3-8B\"\n",
+    "        ),\n",
+    "        resume_if_exists=True,\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "### NeMo 2.0 finetun API\n",
+    "Using all the components we created above, we can call NeMo 2.0 finetun API:\n",
+    "```\n",
+    "llm.finetune(\n",
+    "    model=llama3_8b(),\n",
+    "    data=squad(),\n",
+    "    trainer=trainer(),\n",
+    "    peft=lora(),\n",
+    "    log=logger(),\n",
+    "    optim=adam_with_cosine_annealing(),\n",
+    "    resume=resume(),\n",
+    ")\n",
+    "```\n",
+    "Below is a python script that you can save as a file e.g. `nemo2-peft.py`, and run PEFT training, using all components we created above and NeMo 2.0 finetune API. The script cannot be directly executed in interactive environment like a notebook. We can execute by `python nemo2-peft.py` if single GPU is used, or `torchrun --nproc_per_node=<NUM_GPU> nemo2-peft.py` if multiple GPU is used."
```

Comment:
fix punctuation, fix capitalziation, revise sentences

   "Below is a Python script that you can save as a file, e.g., nemo2-peft.py, to run PEFT training using all the components we created above and the NeMo 2.0 fine-tune API. The script cannot be directly executed in an interactive environment like a notebook. We can execute it by running python nemo2-peft.py if a single GPU is used, or torchrun --nproc_per_node= nemo2-peft.py if multiple GPUs are used."

Reply:
Oops, I actually need to delete this one paragraph since we are no longer using python API but rather nemo-run.